### ISSUE TITLE
function-naming false positive

### ISSUE THREAD

**Issue by vanniktech** [pinterest/ktlint#2018]
ktlint 0.49.0 and this one:

```
import android.location.Location as AndroidLocation

fun AndroidLocation(
  provider: String,
  latitude: Double,
  longitude: Double,
  time: Instant,
) = AndroidLocation(provider).apply {
  this.latitude = latitude
  this.longitude = longitude
  this.time = time.toEpochMilliseconds()
}
```

yields: 

> Function name should start with a lowercase letter (except factory methods) and use camel case (standard:function-naming)

My understanding is that my function is a factory method though and hence it should be fine.

**Comment by paul-dingemans** [comment]
Ktlint requires factory methods to have a declared type. This is loosely based on https://kotlinlang.org/docs/coding-conventions.html#function-names:
```
    @Test
    fun `Given a factory method then do not emit`() {
        val code =
            """
            interface Foo
            class FooImpl : Foo
            fun Foo(): Foo = FooImpl()
            fun Bar.Foo(): Foo = FooImpl()
            """.trimIndent()
        functionNamingRuleAssertThat(code).hasNoLintViolations()
    }
```

**Comment by vanniktech** [comment]
Fair enough 👍
